python/lsst/pipe/tasks/dataFrameActions/_evalColumnExpression.py

```diff
@@ -85,6 +85,8 @@ def makeColumnExpressionAction(className: str, expr: str,
                                                                          Type[DataFrameAction]]]] = None,
                                docstring: str = None
                                ) -> Type[DataFrameAction]:
+    import inspect
+    new_module = inspect.stack()[1].frame.f_locals['__name__']
```

Comment:
Eek, is this the kind of thing we do elsewhere in pex_config to get module names?

If so, pex_config has made me sad today.  If not, I'm not gonna get in the way, but my degree of caution about using `ConfigurableAction` more broadly (and hope that someday we'll be able to replace it) will increase.

In either case, could you add a check for this evaluating to `__main__` and raise an exception if that happens, since otherwise that'll lead to more confusing later exception?

Reply:
I thought about the need to add a comment on this while falling asleep, and forgot by the time I woke up. I will clarify here, this **is not** a generic ``ConfigurableAction`` feature. This function is a factory for creating ConfigurableActions based on an expression defined by a string (see [here](https://github.com/lsst/pipe_tasks/blob/main/python/lsst/pipe/tasks/dataFrameActions/_actions.py#L112) )

The problem this solves is that the class thinks its namespace resolves to inside the function body when pex_config asks, however the actual import path is the module that calls this factory to produce the action. This is not dissimilar to namedtuples.